mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraApplicableFlagDAO.java

```diff
@@ -75,36 +71,26 @@ private PreparedStatement prepareSelect(Session session) {
                 rowOptional.map(row -> new FlagsExtractor(row).getApplicableFlags()));
     }
 
-    public CompletableFuture<Void> updateApplicableFlags(CassandraId cassandraId, Flags flags) {
-        Flags newFlags = new FlagsBuilder().add(flags).build();
-        newFlags.remove(Flag.RECENT);
-        newFlags.remove(Flag.USER);
+    public CompletableFuture<Void> updateApplicableFlags(CassandraId cassandraId, Flags oldFlags, Flags newFlags) {
```

Comment:
Don't we have a class for that 'showing flag delta' ? If no maybe we should introduce one.
Imagine someone revert the order of oldFlags and newFlags the compilator would not lie